Formula/imessage-exporter.rb

```diff
@@ -0,0 +1,19 @@
+class ImessageExporter < Formula
+  desc "Export MacOS iMessage data + run iMessage Diagnostics"
+  homepage "https://github.com/ReagentX/imessage-exporter"
+  url "https://github.com/ReagentX/imessage-exporter/archive/refs/tags/0.1.7.tar.gz"
+  sha256 "0d54c614692795bf34a6e159c6e0eb7a61856aca0d72ce6efe7665c3b686ae6e"
+  license "GPL-3.0-or-later"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--path", "imessage-exporter", "--locked", "--root", prefix
+    bin.install "target/release/imessage-exporter"
+  end
+
+  test do
+    system "imessage-exporter -h"
+    system "echo", "Done!"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
this software exports iMessage data, without providing someone's iMessage database as part of the distribution to there would be nothing to test

Reply:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
> Same if the software requires a virtual machine, docker instance, etc. to be running.

Same applies for "software needs data we can not provide"

Reply:
If you want a test to see if it runs without any provided data, then the current test performs that task.

Reply:
For one I don't think we need to test the ability of `echo`. And second, it currently tests the help message.
> We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

Reply:
The program doesn't do anything without and iMessage database and I don't see abuse rushing to upload theirs.

Reply:
> The program doesn't do anything without an iMessage database and I don't see anyone rushing to upload theirs.

So assert that it doesn't do anything if you ask it to export nothing.

> I can add some tests for various CLI arg combos that fail, but that isn't really any more useful test than the current one. 

Help flags rarely execute the same code path as the actual program so as a package manager we disagree with the idea that it's not any more useful since a failure to read a database is more likely to throw dependency errors than a help text.

> I can remove the echo but other than that there isn't much I can do here.

You should definitely remove the echo since that's not part of imessage-exporter.